### PR TITLE
fix short living lki bug for Chromatic Star

### LINKED ISSUE #12330: Chromatic Star and Serra Paragon interaction is bugged
Start with [[Chromatic Star]] on the graveyard and [[Serra Paragon]] in play. Cast Star from the graveyard, then sacrifice it. The added trigger is not trigger (card draw from Star is).

image

This discard seems to happen before the trigger is checked:
image

As further info that may lead to the bug source, [[Haywire Mite]] works properly:
image

### PR DESCRIPTION
**Pull request by Susucre**
So the Chromatic Star's trigger was not using short living lki properly, and for some reason that prevented the trigger gained from Paragon to trigger.

Question: should some of the following Trigger also be set up with short living lki the same way?
image

fix #12330

### CODE REVIEW

**Comment by JayDi85** [comment]
Yes, all dies like triggers (put into a graveyard from the battlefield) must use short lki logic

**Comment by Susucre** [comment]
I don't think it should be added for the FromAnywhere triggers. That does break a few unit tests in a bad way. Makes sense however for the PutIntoGraveFromBattlefieldAllTriggeredAbility one.

File: Mage/src/main/java/mage/abilities/common/PutIntoGraveFromBattlefieldAllTriggeredAbility.java [hunk 1]
```diff
@@ -63,4 +63,11 @@ public boolean checkTrigger(GameEvent event, Game game) {
     public PutIntoGraveFromBattlefieldAllTriggeredAbility copy() {
         return new PutIntoGraveFromBattlefieldAllTriggeredAbility(this);
     }
+
+    // Not sure if needed?
+    //
+    //@Override
```
Comment (Susucre) [inline Mage/src/main/java/mage/abilities/common/PutIntoGraveFromBattlefieldAllTriggeredAbility.java:69]: I'll try to find a card with the trigger that can come back with Serra Paragon. My current test with Aetherwork Marvel does work on master code.

Comment (JayDi85) [reply Mage/src/main/java/mage/abilities/common/PutIntoGraveFromBattlefieldAllTriggeredAbility.java:69]: You can try to return false in `getShortLivingLKI` to see broken tests and find cards/effects that can potentially reproduce bad use cases for such triggers.

**Review by JayDi85 — APPROVED** [review]
Anyway it's equal to dies trigger so must use same code. No needs in tests for each use cases or card.

**Comment by xenohedron** [comment]
Well, there are more differences...
- `DiesSourceTriggeredAbility extends ZoneChangeTriggeredAbility`
- `PutIntoGraveFromBattlefieldSourceTriggeredAbility extends TriggeredAbilityImpl`

Anyway, good test coverage on dies triggers will help with any rework of the short living lki logic.

**Pull request merged**